pvlib/tests/ivtools/test_sdm.py

```diff
@@ -80,6 +80,26 @@ def test_fit_desoto():
                        rtol=1e-4)
 
 
+def test_fit_desoto_init_guess():
+    init_guess = {'IL_0': 9.43, 'Io_0': 6.04e-7, 'Rs_0': 0.155, 'Rsh_0': 100.,
+                  'a_0': 2.312}
+    result, _ = sdm.fit_desoto(v_mp=31.0, i_mp=8.71, v_oc=38.3, i_sc=9.43,
+                               alpha_sc=0.005658, beta_voc=-0.13788,
+                               cells_in_series=60, init_guess=init_guess)
+    result_expected = {'I_L_ref': 9.45232,
+                       'I_o_ref': 3.22460e-10,
+                       'R_s': 0.297814,
+                       'R_sh_ref': 125.798,
+                       'a_ref': 1.59128,
+                       'alpha_sc': 0.005658,
+                       'EgRef': 1.121,
+                       'dEgdT': -0.0002677,
+                       'irrad_ref': 1000,
+                       'temp_ref': 25}
```

Comment:
These expected values are the same as when the default starting point is used, meaning this test doesn't ensure that supplying the `init_guess` parameter has any effect.  Rather than testing expected values, I think we should use a mocker and assert that `optimize.root` was `called_once_with(x0=np.array([...]))` (https://pvlib-python.readthedocs.io/en/stable/contributing/testing.html#pvsystem-location)

Reply:
Got it to work as suggested. Some finicky details are necessary: the double splat fourth argument to get the same number of arguments, and wrapping the `x0` numpy array with `pytest.approx`. Those may be worth adding as notes to the contributing example.